tests/sentry/integrations/slack/webhooks/events/test_discover_link_shared.py

```diff
@@ -68,6 +104,60 @@ def share_discover_links(self, mock_match_link, mock_):
         data = responses.calls[0].request.body
         return dict(parse_qsl(data))
 
+    @patch(
+        "sentry.integrations.slack.webhooks.event.match_link",
+        # match_link will be called twice, for each our links. Resolve into
+        # two unique links and one duplicate.
+        side_effect=[
+            (LinkType.DISCOVER, {"arg1": "value1"}),
+            (LinkType.DISCOVER, {"arg1", "value2"}),
+            (LinkType.DISCOVER, {"arg1": "value1"}),
+        ],
+    )
+    @patch("sentry.integrations.slack.requests.event.has_discover_links", return_value=True)
+    @patch(
+        "sentry.integrations.slack.webhooks.event.link_handlers",
+        {
+            LinkType.DISCOVER: Handler(
+                matcher=[re.compile(r"test")],
+                arg_mapper=make_type_coercer({}),
+                fn=Mock(return_value={"link1": "unfurl", "link2": "unfurl"}),
+            )
+        },
+    )
```

Comment:
if you have the same patch decorators for most functions in a test class, you can move them to the top of the class and it will apply to every function

Reply:
ill do this when i ga and delete the old client tests